src/test/scala/chiselTests/BlackBox.scala

```diff
@@ -222,4 +230,48 @@ class BlackBoxSpec extends ChiselFlatSpec {
       })
     }).getMessage should include("must have a port named 'io' of type Record")
   }
+
+  "BlackBoxes" should "sort the verilog output of their param map by param key" in {
+    val rand = scala.util.Random
+    rand.setSeed(0)
+
+    class Top(m: Map[String, Param]) extends Module {
+      val io = IO(new Bundle {})
+      val ibufds = Module(new ParameterizedBlackBox(m))
+      ibufds.io.I := clock
+    }
+
+    def keysInOrder(s: String): Boolean = {
+      val nums = s.split(", ").map { ss => ss.drop(5).take(3) }
+      nums.zip(nums.tail).forall { case (a, b) => a < b }
+    }
+
+    // make a list of keys / param pairs and add them as they
+    // are generated in random order
+    // rendering verilopg will
+    for (_ <- 0 until 16) {
+      val numberOfKeys = rand.nextInt(32)
+      val paramMap = (0 until numberOfKeys).map {
+        case _ =>
+          val keyNumber = rand.nextInt(48)
+          f"key_$keyNumber%03d" -> IntParam(keyNumber)
+      }
+
+      val verilogString = getVerilogString(new Top(paramMap.toMap))
+
+      var paramsSeen = false
+      val paramsShowInOrder = verilogString.split("\n").exists { line =>
+        val offset = line.indexOf(".key_0")
+        val result = if (offset >= 0) {
+          paramsSeen = true
+          keysInOrder(line.drop(offset))
+        } else {
+          false
+        }
+        result
+      } || !paramsSeen
```

Comment:
I find this logic pretty hard to follow. This is a test so it doesn't matter _that_ much but I think it would be more clear and still sufficient to just take a static list of like 6 parameters like parameter named `A` has value `0`, `B` has value `1`, etc., randomize there order, and then just check the exactly expected line in the Verilog, something like:

```scala
verilogString should include ("#(.A(0), .B(1), .C(2)...")
```

Reply:
Makes sense, I've simplified the test